### PR TITLE
[refactor]: introduce full-featured code groups

### PR DESCRIPTION
**Pull request by 0x009922**
## Example

````md
:::code-group

```
Plain fence
```

```ts
// with language
type S = string
```

# Add any-level heading to make custom title

```
Lorem
```

# Import snippets

<<<@/snippets/anything.ts

:::
````

https://user-images.githubusercontent.com/43530070/201253080-bc0f55cb-fbe8-4f3f-b7ce-1bbfbfd67d7e.mov

## Features

- Code groups for fense blocks (including builtin `<<<` snippet imports)
- Block title via a leading heading of any level (`#`, `##` and so on)
- Accessible - implemented according to [Tabs](https://www.w3.org/WAI/ARIA/apg/patterns/tabpanel/) design pattern, i.e. supports navigation through <kbd>Tab</kbd>, <kbd>→</kbd> and <kbd>←</kbd> and is marked by ARIA attributes.

## Reverts

- Old snippets syntax is removed. Not a big change because it wasn't used anywhere except the snippets guide. It could be replaced with code groups.
  
  **Before**

  ````md
  ::snippets
  debug_python_Lorem.py
  ::
  ````

  **After**

  ```md
  :::code-group

  # Python
  
  <<<@/snippets/debug_python_Lorem.py

  :::
  ```
- No tabs scrolling and touch support. It could be added, but the visual design should be discussed first - maybe scrolling will not be needed at all.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -89,6 +89,22 @@ To fix auto-fixable issues, run:
 pnpm lint --fix
 ```
 
+### Testing
+
+We use [Vitest](https://vitest.dev/) test framework to assure quality of non-trivial internal parts of the project.
+
+To check whether tests passes, run:
```
Comment (WRRicht3r) [inline README.md:96]:
```suggestion
To check whether tests pass, run:
```

File: src/documenting/code-groups.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+<<<@/snippets/debug_java_Lorem.java
+
+:::
+
+**Rules:**
+
+- All fences (i.e. code blocks) within a `code-group` container are
+  considered as dedicated tabs;
+- If fence has a heading before it, then the heading content is used as a
```
Comment (WRRicht3r) [inline src/documenting/code-groups.md:62]:
```suggestion
- If the fence has a heading before it, then the heading content is used as a
```

File: src/documenting/code-groups.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+  title. Heading could have any level (i.e. `#`, `##` and so on).
```
Comment (WRRicht3r) [inline src/documenting/code-groups.md:63]:
```suggestion
  title. The heading could have any level (i.e. `#`, `##`, and so on).
```

File: src/documenting/code-groups.md [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+- If fence doesn't have a title, then it's language is used as a title. If
```
Comment (WRRicht3r) [inline src/documenting/code-groups.md:64]:
```suggestion
- If the fence doesn't have a title, then it's language is used as a title. If
```

File: src/documenting/code-groups.md [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+  there is no language, then titles fallbacks to `<block-#i>` placeholder.
```
Comment (WRRicht3r) [inline src/documenting/code-groups.md:65]:
```suggestion
  there is no language, then the title falls back to the `<block-#i>` placeholder.
```

File: src/documenting/code-groups.md [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+
+## HTML-based syntax
+
+It is more low-level, but might be used for advanced cases.
```
Comment (WRRicht3r) [inline src/documenting/code-groups.md:69]:
```suggestion
It is lower-level but might be used for advanced cases.
```

File: src/documenting/snippets.md [hunk 7]
```diff
@@ -134,144 +136,71 @@ starting with a comment symbol(s). If there's common padding behind each
...
+snippets.
 
-```
-::snippets
-debug_rust_Lorem.rs
-debug_typescript_Lorem.ts
-::
-```
+Keeping in mind that a default snippet directory is `src/snippets/`, a
+snippet could be imported as following:
```
Comment (WRRicht3r) [inline src/documenting/snippets.md:146]:
```suggestion
snippet could be imported as follows:
```

**Review by WRRicht3r — CHANGES_REQUESTED** [review]

**Review by WRRicht3r — APPROVED** [review]

File: src/documenting/code-groups.md [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+### Snippet syntax is supported as well
+
+<<<@/snippets/debug_java_Lorem.java
+
+:::
+
+**Rules:**
+
+- All fences (i.e. code blocks) within a `code-group` container are
+  considered as dedicated tabs;
```
Comment (outoftardis) [inline src/documenting/code-groups.md:61]:
```suggestion
- Each fence (a code block) within a `code-group` container is
  rendered as a tab;
```

File: src/documenting/code-groups.md [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+- If the fence has a heading before it, then the heading content is used as a
```
Comment (outoftardis) [inline src/documenting/code-groups.md:62]:
```suggestion
- If a fence has a heading before it, then the heading content is used as a
```

File: src/documenting/code-groups.md [hunk 10]
```diff
@@ -0,0 +1,165 @@
...
+  title. The heading could have any level (i.e. `#`, `##`, and so on).
+- If the fence doesn't have a title, then it's language is used as a title. If
```
Comment (outoftardis) [inline src/documenting/code-groups.md:64]:
```suggestion
- If a fence doesn't have a title, then its language is used as a title. If
```

File: src/documenting/code-groups.md [hunk 11]
```diff
@@ -0,0 +1,165 @@
...
+Fence without lang or title
+```
+
+### Snippet syntax is supported as well
+
+<<<@/snippets/debug_java_Lorem.java
+
+:::
+
+**Rules:**
```
Comment (outoftardis) [inline src/documenting/code-groups.md:58]: I'd put this before input-output example

Comment (0x009922) [reply src/documenting/code-groups.md]: Done

**Review by outoftardis — APPROVED** [review]

**Pull request merged**